### PR TITLE
BUG: Set index when reading stata file

### LINKED ISSUE #16342: read_stata ignores `index` parameter
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import pandas.util.testing as tm

In [19]: df = tm.makeDataFrame()

In [20]: df.to_stata('foo.stata')

In [21]: pd.read_stata('foo.stata', index='index')
Out[21]:
         index         A         B         C         D
0   2wIkgwAbVm  0.310317 -0.753652 -0.509405 -0.309492
1   59guDFHo0l  0.404983 -0.098995 -0.330786 -0.548584
2   5sX3JbPQjk  0.520377  1.379943  2.375164  1.475831
3   UsF3izOnQy -1.413272  0.668192  0.859581 -0.646165
...
```

It should be like `read_stata('foo.stata').set_index('index')`

Side-note, I think elsewhere we use `index_col` instead of `index`.

#### Output of ``pd.show_versions()``

master

### PR DESCRIPTION
**Pull request by bashtage**
Ensures index is set when requested when reading state dta file

closes #16342

- [x] closes #16342
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/stata.py [hunk 1]
```diff
@@ -1486,6 +1486,8 @@ def read(self, nrows=None, convert_dates=None,
             columns = self._columns
         if order_categoricals is None:
             order_categoricals = self._order_categoricals
+        if index is None:
+            index = self._index
```
Comment (gfyoung) [inline pandas/io/stata.py]: Do we have tests that hit this path?

Comment (bashtage) [inline pandas/io/stata.py]: Added in PR

**Comment by TomAugspurger** [comment]
Did we want to change the keyword from `index` to `index_col` to match `read_csv`?

**Comment by bashtage** [comment]
Probably a good idea.

**Comment by bashtage** [comment]
Does this need deprecation?

**Comment by bashtage** [comment]
Last one, is there a pattern to follow for renaming (e.g. a decorator, *kwargs, etc)

**Comment by TomAugspurger** [comment]
Yeah, there's an `@deprecate_kwarg` decorator somewhere. Honestly not sure about whether it needs a deprecation, since the old one didn't do anything, but we may as well.

**Comment by TomAugspurger** [comment]
It's in `pandas/util/_decorators` btw.

**Comment by bashtage** [comment]
I went ahead and did the insta-deprecate.

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -293,6 +293,7 @@ Other API Changes
 - :func:`Series.argmin` and :func:`Series.argmax` will now raise a ``TypeError`` when used with ``object`` dtypes, instead of a ``ValueError`` (:issue:`13595`)
 - :class:`Period` is now immutable, and will now raise an ``AttributeError`` when a user tries to assign a new value to the ``ordinal`` or ``freq`` attributes (:issue:`17116`).
 - :func:`to_datetime` when passed a tz-aware ``origin=`` kwarg will now raise a more informative ``ValueError`` rather than a ``TypeError`` (:issue:`16842`)
+- Renamed non-functional `index` to `index_col` in :func:`read_stata` to improve API consistency (:issue:`16342`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: can you use double backticks here (around index,index_col)

File: pandas/tests/io/test_stata.py [hunk 3]
```diff
@@ -1309,3 +1309,11 @@ def test_value_labels_iterator(self, write_index):
             dta_iter = pd.read_stata(path, iterator=True)
             value_labels = dta_iter.value_labels()
         assert value_labels == {'A': {0: 'A', 1: 'B', 2: 'C', 3: 'E'}}
+
+    def test_set_index(self):
+        df = tm.makeDataFrame()
```
Comment (jreback) [inline pandas/tests/io/test_stata.py]: can you add the issue number

**Review by jreback — CHANGES_REQUESTED** [review]
minor comments. can you deprecate ``index`` (and direct to ``index_col``) as well.

**Comment by TomAugspurger** [comment]
> minor comments. can you deprecate index (and direct to index_col) as well.

Do we need a deprecation, or can we just "break" API here. `index=...` didn't work at all earlier, so I'm OK with just changing without a deprecation cycle.

**Comment by bashtage** [comment]
@jreback @TomAugspurger I think this is GTG

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**